### PR TITLE
Ensure subprocesses are run via Runtime()

### PR DESCRIPTION
**Pull request by ssbarnea**
Instead of directly calling subprocess, ensure we use Runtime
api methods for running various commands, most importantly
ansible commands.

This also removes subprocess-tee dependency as a desired action.

### CODE REVIEW

**Comment by webknjaz** [comment]
Idea: we could forbid calling subprocess via pylint.

**Comment by ssbarnea** [comment]
> Idea: we could forbid calling subprocess via pylint.
We can do this in follow-up, I am trying to make this PR as small as possible.

File: src/molecule/app.py
Comment (webknjaz) [inline src/molecule/app.py:13]: Is this an attempt to make a mutable singleton? Why not initialize it during the startup sequence and then pass around DI-style?

File: setup.cfg
Comment (webknjaz) [inline setup.cfg:62]: Let's be stricter until the API is stable:
```suggestion
    ansible-compat ~= 1.0.0a0
```

File: src/molecule/app.py
Comment (ssbarnea) [reply src/molecule/app.py:13]: Molecule multi-layered structure makes it extremely inconvenient to pass such an object. I considered that approach but it would be a total mess. We can aim to achieve that at some point but it would need several major version changes and existence of plugins is a major limiting factor.

Comment (webknjaz) [reply src/molecule/app.py:13]: Affecting the import time is also messy. There's nothing wrong with multilayered bypasses, though.

**Comment by ssbarnea** [comment]
@webknjaz @ganeshrn: I do not plan to merge this until I release 1.0.0 or ansible-compat, but the release of compat is waiting for the approval of this PR as I want to avoid having to patch it in case we miss something.

Please review it and let me know if I can release compat.

File: src/molecule/app.py
Comment (ssbarnea) [reply src/molecule/app.py:13]: I will not do that as part of this PR. If you want you can lead this challenge but I warn you that you will unlikely to succeed as you will have to change API of plugins and 15+ plugins. I am also worried that this may add delays which may interfere with `AAP` inclusion goals.

**Review by tadeboro — APPROVED** [review]

**Pull request merged**
